ui/tests/acceptance/secrets/backend/database/secret-test.js

```diff
@@ -480,6 +484,19 @@ module('Acceptance | secrets/database/*', function (hooks) {
     assert.dom('[data-test-get-credentials]').isEnabled();
   });
 
+  test('connection_url must be decoded', async function (assert) {
+    const backend = this.backend;
+    const connection = await newConnection(
+      backend,
+      'mongodb-database-plugin',
+      '{{username}}/{{password}}@oracle-xe:1521/XEPDB1'
+    );
+    await navToConnection(backend, connection);
+    assert
+      .dom('[data-test-row-value="Connection URL"]')
+      .hasText('{{username}}/{{password}}@oracle-xe:1521/XEPDB1');
+  });
```

Comment:
Just to verify, if you remove the `decode-uri` helper from the template this test fails?

Reply:
<img width="1615" alt="Screenshot 2023-10-17 at 10 34 33 AM" src="https://github.com/hashicorp/vault/assets/30884335/75bff71c-4a40-4954-a135-8960e6e25012">
Yup it fails 😊

Reply:
Perfect thanks for adding this test!